pkg/compass/compass.go

```diff
@@ -174,13 +174,39 @@ func PublicRaftMultiaddr() (multiaddr.Multiaddr, error) {
 	if ip.To4() == nil {
 		ipVersion = "ip6"
 	}
-	addr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/%s/%s/tcp/%s/p2p/%s", ipVersion, ip.String(), p2pPort, PeerID().String()))
+	addr, err := multiaddr.NewMultiaddr(fmt.Sprintf("/%s/%s", ipVersion, ip.String()))
 	if err != nil {
 		return nil, err
 	}
 	return addr, nil
 }
 
+func IPFSSwarmAddr() (multiaddr.Multiaddr, error) {
+	ipMA, err := IPMultiaddr()
+	if err != nil {
+		return nil, err
+	}
+	tail, err := multiaddr.NewMultiaddr(fmt.Sprintf("/tcp/4001/p2p/%s", PeerID().String()))
```

Comment:
is it always 4001?

Reply:
Nope. I'll fix

Reply:
Actually can we handle this in a follow up PR? IPFS reconfiguration was pretty broken last time I checked and is gonna take some extra time to debug